rectools/models/nn/item_net.py

```diff
@@ -88,36 +93,26 @@ def forward(self, items: torch.Tensor) -> torch.Tensor:
         torch.Tensor
             Item embeddings.
         """
-        feature_dense = self.get_dense_item_features(items)
-
-        feature_embs = self.category_embeddings(self.feature_catalog.to(self.device))
-        feature_embs = self.drop_layer(feature_embs)
-
-        feature_embeddings_per_items = feature_dense.to(self.device) @ feature_embs
+        item_emb_bag_inputs, item_offsets = self.get_item_inputs_offsets(items)
+        feature_embeddings_per_items = self.embedding_bag(input=item_emb_bag_inputs, offsets=item_offsets)
+        feature_embeddings_per_items = self.drop_layer(feature_embeddings_per_items)
         return feature_embeddings_per_items
 
+    def get_item_inputs_offsets(self, items: torch.Tensor) -> tp.Tuple[torch.Tensor, torch.Tensor]:
+        """Get categorical item features and offsets for `items`."""
+        item_indexes = self.offsets[items].unsqueeze(-1) + self.emb_bag_indexes
+        length_mask = self.emb_bag_indexes < self.len_indexes[items].unsqueeze(-1)
+        item_emb_bag_inputs = self.emb_bag_inputs[item_indexes[length_mask]].squeeze(-1)
+        item_offsets = torch.cat(
+            (torch.tensor([0], device=self.device), torch.cumsum(self.len_indexes[items], dim=0)[:-1])
+        )
+        return item_emb_bag_inputs, item_offsets
```

Comment:
Let's process corner cases:
1. Only one item in `forward` and it has length 1 => incorrect outputs
2. Forward pass on multiple items where one of them has NO features.